src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+        }
+      }
+
+      // ---------- Pools ----------
+      let pools = await this.getPoolIdentifiers(
+        srcToken,
+        destToken,
+        side,
+        blockNumber,
+      );
+      if (pools.length === 0) return null;
+
+      // ---------- Prices ----------
+      const priceMap = await this.getCachedPrices();
+
+      if (!priceMap) return null;
+
+      let isInputQuote = false;
+      let pairKey = `${normalizedSrcToken.symbol}/${normalizedDestToken.symbol}`;
+      const pairsKeys = Object.keys(priceMap);
+
+      if (!pairsKeys.includes(pairKey)) {
+        // Revert
+        isInputQuote = true;
+        pairKey = `${normalizedDestToken.symbol}/${normalizedSrcToken.symbol}`;
+        if (!pairsKeys.includes(pairKey)) {
+          return null;
+        }
+      }
+
+      /**
+       * Orderbook
+       */
+      const priceData = priceMap[pairKey];
+
+      let orderbook: any[] = [];
+      if (side === SwapSide.BUY) {
+        orderbook = priceData.asks;
+      } else {
+        orderbook = priceData.bids;
+      }
+      if (orderbook?.length === 0) {
+        throw new Error(`Empty orderbook for ${pairKey}`);
+      }
+
+      const prices = this.calculateOrderPrice(
+        amounts,
+        orderbook,
+        side === SwapSide.SELL ? srcToken : destToken,
+        side === SwapSide.SELL ? destToken : srcToken,
+        side === SwapSide.SELL ? isInputQuote : !isInputQuote,
+      );
+
+      const result = [
+        {
+          prices: prices,
+          unit: BigInt(normalizedDestToken.decimals),
+          exchange: this.dexKey,
+          gasCost: CABLES_GAS_COST,
+          poolAddresses: [this.mainnetRFQAddress],
+          data: {},
+        },
+      ];
+
+      return result;
+    } catch (e: unknown) {
+      this.logger.error(
+        `Error in getPricesVolume`,
+        {
+          srcToken: srcToken.address || srcToken.symbol,
+          destToken: destToken.address || destToken.symbol,
+          side,
+        },
+        e,
+      );
+      return null;
+    }
+  }
+
+  getCalldataGasCost(poolPrices: PoolPrices<CablesData>): number | number[] {
+    return (
+      CALLDATA_GAS_COST.DEX_OVERHEAD +
+      // addresses: makerAsset, takerAsset, maker, taker
+      CALLDATA_GAS_COST.ADDRESS * 4 +
+      // uint256: expiry
+      CALLDATA_GAS_COST.wordNonZeroBytes(16) +
+      // uint256: nonceAndMeta, makerAmount, takerAmount
+      CALLDATA_GAS_COST.AMOUNT * 3 +
+      // bytes: _signature (65 bytes)
+      CALLDATA_GAS_COST.FULL_WORD * 2 +
+      CALLDATA_GAS_COST.OFFSET_SMALL
+    );
+  }
+
+  async initializePricing(blockNumber: number): Promise<void> {
+    if (!this.dexHelper.config.isSlave) {
+      this.rateFetcher.start();
+    }
+    return;
+  }
+
+  getAdapters(side: SwapSide): { name: string; index: number }[] | null {
+    return null;
+  }
+
+  releaseResources?(): AsyncOrSync<void> {
+    if (this.rateFetcher) {
```

Comment:
looks like `this.rateFetcher` is always present and better to replace `!this.dexHelper.config.isSlave`, just because polling works only on `isSlave = false`